getting-started/dotnet/00.MIGRATION.md

```diff
@@ -0,0 +1,166 @@
++
++    public EchoBot(ApplicationOptions<TurnState, TurnStateManager> options) : base(options) {}
++ }
+```
+
+> The `TurnState` and `TurnStateManager` are classes that make up the turn state infrastructure. The generic types in the `Application` class allow users to customize the shape of the turn state.
+
+The activity handling method is the same for both (previous) `TeamsActivityHandler` and (new) `Application` class, except for a few nuances. See below for more.
+
+#### New turn state parameter:
+
+`TState turnState` has been added as a parameter to `OnMessageActivityAsync`.
+
+```diff
+protected virtual Task OnMessageActivityAsync(
+    ITurnContext<IMessageActivity> turnContext,
++   TState turnState,
+    CancellationToken cancellationToken);
+```
+
+#### Drop the `Teams` prefix:
+
+`onTeamsChannelCreatedAsync` is modified to `OnChannelCreatedAsync`.
+
+```diff
+protected virtual Task
+-    OnTeamsChannelCreatedAsync(
++    OnChannelCreatedAsync(
+        ChannelInfo channelInfo,
+        TeamInfo teamInfo,
+        ITurnContext<IConversationUpdateActivity> turnContext,
+        TState turnState,
+        CancellationToken cancellationToken);
+```
+
+#### Reorder parameters for consistency:
+
+> Note that in `OnSearchInvokeAsync`, the parameter `SearchInvokeValue: invokeValue` has been moved to the first parameter, to precede `ITurnContext<IInvokeActivity>: turnContext`. `CancellationToken cancellationToken` remains as the last parameter.
+
+```diff
+protected virtual Task<SearchInvokeResponse> OnSearchInvokeAsync(
+      SearchInvokeValue invokeValue,
+      ITurnContext<IInvokeActivity> turnContext,
++     TState turnState,
+      CancellationToken cancellationToken);
+```
+
+For every activity handler method in the BotFramework SDK, users can replace it with a corresponding method in the `Application` class.
+
+This is all it takes to port over to Teams AI!
+
+Next, the exciting step is to add AI to your bot! Continue on to [01.AI-SETUP](01.AI-SETUP.md)
+
+Otherwise, see below for in-depth explanation of porting-related concepts.
+
+## How an incoming activity is routed in the Application:
+
+When an incoming activity reaches the server, the bot adapter handles the necessary authentication and creates a `TurnContext` object that encapsulates the activity details. It then calls the `OnTurnAsync` method. This is the entry point method of the application. Here's what happens in this method:
+
+1. If configured in the application options, pulses of the `Typing` activity are sent to the user.
+2. If configured in the application options, the @mention is removed from the incoming message activity.
+3. The turn state is loaded using the configured turn state manager.
+4. The `OnBeforeTurnAsync` activity handler is executed. If it returns false, save turn state to storage.
+5. If the corresponding activity handler is configured, then it is executed. Otheriwse the AI module's `ChainAsync` executes.
```

Comment:
Diff from JS' fluent API (`app.message("some-pattern", ...)`), in .NET SDK all text messages go to `OnMessageActivityAsync`. Then it may require calling `base.OnMessageActivityAsync` (or actually throw a `NotImplementedException`) to let AI to handle remaining messages. Otherwise all messages are short-cut and no AI action is executed.

See the code [here](https://github.com/microsoft/teams-ai/blob/DOTNET/dotnet/samples/04.ai.d.chainedActions.listBot/ListBotApplication.cs#L68) in sample.

I did met this potential issue when adding sample. Do we need to call out this somewhere in .NET doc?

Reply:
Thanks for the example and correction. It would be great to have more documentation on this if you have suggestions.